src/app/app-platform/ContentAppPlatform.cpp

```diff
@@ -108,38 +110,102 @@ EndpointId ContentAppPlatform::AddContentApp(ContentApp * app, EmberAfEndpointTy
     index = 0;
     while (index < CHIP_DEVICE_CONFIG_DYNAMIC_ENDPOINT_COUNT)
     {
-        if (nullptr == mContentApps[index])
+        if (mContentApps[index] != nullptr)
         {
-            mContentApps[index] = app;
-            EmberAfStatus ret;
-            while (1)
+            index++;
+            continue;
+        }
+        mContentApps[index] = app;
+        EmberAfStatus ret;
+        EndpointId initEndpointId = mCurrentEndpointId;
+
+        do
+        {
+            ret = emberAfSetDynamicEndpoint(index, mCurrentEndpointId, ep, dataVersionStorage, deviceTypeList);
+            if (ret == EMBER_ZCL_STATUS_SUCCESS)
             {
-                ret = emberAfSetDynamicEndpoint(index, mCurrentEndpointId, ep, dataVersionStorage, deviceTypeList);
-                if (ret == EMBER_ZCL_STATUS_SUCCESS)
-                {
-                    ChipLogProgress(DeviceLayer, "Added ContentApp %s to dynamic endpoint %d (index=%d)", vendorApp.applicationId,
-                                    mCurrentEndpointId, index);
-                    app->SetEndpointId(mCurrentEndpointId);
-                    return app->GetEndpointId();
-                }
-                else if (ret != EMBER_ZCL_STATUS_DUPLICATE_EXISTS)
-                {
-                    ChipLogError(DeviceLayer, "Adding ContentApp error=%d", ret);
-                    return kNoCurrentEndpointId;
-                }
-                // Handle wrap condition
-                if (++mCurrentEndpointId < mFirstDynamicEndpointId)
-                {
-                    mCurrentEndpointId = mFirstDynamicEndpointId;
-                }
+                ChipLogProgress(DeviceLayer, "Added ContentApp %s to dynamic endpoint %d (index=%d)", vendorApp.applicationId,
+                                mCurrentEndpointId, index);
+                app->SetEndpointId(mCurrentEndpointId);
+                IncrementCurrentEndpointID();
+                return app->GetEndpointId();
             }
+            else if (ret != EMBER_ZCL_STATUS_DUPLICATE_EXISTS)
+            {
+                ChipLogError(DeviceLayer, "Adding ContentApp error=%d", ret);
```

Comment:
we are setting mContentApps[index]. Should that set be reverted?

Reply:
Moved setting the entry in the content apps array only to the success case.